tests/test_message/test_message_create.py

```diff
@@ -98,19 +162,64 @@ async def test_message_create_failure(
     assert response.status_code == 502, response.text
 
 
-async def test_message_create_400(client, db_service, request_data, request_headers):
-    request_headers["X-Hub-Signature-256"] = ""
-    response = await client.post(
-        f"/api/v1/messages/{db_service.uuid}", content=request_data, headers=request_headers
-    )
+@pytest.mark.parametrize(
+    "kind, request_data, db_service, request_headers",
+    [
+        pytest.param(
+            "github",
+            "github",
+            "github",
+            "github",
+            id="GitHub - Sending data with wrong information",
+        ),
+        pytest.param(
+            "forgejo",
+            "forgejo",
+            "forgejo",
+            "forgejo",
+            id="Forgejo - Sending data with wrong information",
+        ),
+    ],
+    indirect=["request_data", "db_service", "request_headers"],
+)
+async def test_message_create_400(client, db_service, request_data, request_headers, kind):
+    funcpath = f"webhook_to_fedora_messaging.endpoints.parser.{kind}.validate_checksum"
+    with patch(funcpath) as mock_function:
+        mock_function.side_effect = SignatureMatchError("Message signature could not be matched")
+        response = await client.post(
+            f"/api/v1/messages/{db_service.uuid}", content=request_data, headers=request_headers
+        )
     assert response.status_code == 400
```

Comment:
I seem to have run into a wall here.

```
...
FAILED tests/test_message/test_message_create.py::test_message_create_400[GitHub - Sending data with wrong information] - cashews.exceptions.NotConfiguredError: run `cache.setup(...)` before using cache
FAILED tests/test_message/test_message_create.py::test_message_create_400[Forgejo - Sending data with wrong information] - assert 502 == 400 +  where 502 = <Response [502 Bad Gateway]>.status_code
...
```

Am I mocking the side effect for the decorator incorrectly?

I tried digging deep into it, but I could not quite crack it.

Reply:
Yeah that won't work because `validate_checksum` is not a decorator, and has already been called when the module was imported to wrap the parser function. You can no longer mock it.
You can however modify the `request_headers` you are sending to garble the signature and get the same result, no? (the less mocking the better)

Reply:
@nphilipp, thanks for your suggestion on breaking the "forgejo_parser" in order to invoke the `SignatureMatchError`, and in turn, the `400 Bad Request`. I will resort to using that if the `v1.1.1` of `Webhook To Fedora Messaging Messages` does not fix it.

Reply:
@abompard, garbling the signature sounds good too. I was thinking of mocking `hmac.compare_digest` with a side effect to invoke the `SignatureMatchError` exception but you are right - less mocking is good.